datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs

```diff
@@ -381,15 +381,28 @@ macro_rules! get_min_max_values_for_page_index {
                     vec.iter().map(|x| x.$func().cloned()),
                 )))
             }
-            Index::BYTE_ARRAY(index) => {
-                let vec = &index.indexes;
-                let array: StringArray = vec
-                    .iter()
-                    .map(|x| x.$func())
-                    .map(|x| x.and_then(|x| std::str::from_utf8(x).ok()))
-                    .collect();
-                Some(Arc::new(array))
-            }
+            Index::BYTE_ARRAY(index) => match $self.target_type {
```

Comment:
I wonder if this code is covered? Is there some test that does page filtering on Decimal values?

Reply:
No, because there is no UT for the the `page filter`, but I can add some test cases in the follow up pr for the page filter.

Reply:
https://github.com/apache/arrow-datafusion/issues/4791